test/blockgaslimit.js

```diff
@@ -0,0 +1,73 @@
+const Web3 = require("web3");
+const assert = require("assert");
+const Ganache = require("../index");
+const path = require("path");
+const compileAndDeploy = require("./helpers/contracts").compileAndDeploy;
+
+const mnemonic = "candy maple cake sugar pudding cream honey rich smooth crumble sweet treat";
+
+function setUp(options = { mnemonic }, contractName = "BlockGasLimit") {
+  let context = {
+    options: options,
+    provider: null,
+    web3: null,
+    accounts: [],
+    contractArtifact: {},
+    instance: null
+  };
+
+  before("setup web3", async function() {
+    // eslint-disable-next-line new-cap
+    context.provider = new Ganache.provider(context.options);
+
+    // context.provider = new Web3(new Web3.providers.HttpProvider("http://localhost:8545"));
+    context.options.blockTime = 2000;
+    context.web3 = new Web3(context.provider);
+  });
+
+  before("compile source", async function() {
+    this.timeout(10000);
+    context.contractArtifact = await compileAndDeploy(path.join(__dirname, ".", `${contractName}.sol`),
+      contractName,
+      context.web3
+    );
+    context.instance = context.contractArtifact.instance;
+  });
+
+  return context;
+}
+
+describe("Exceeding block gas limitations", function() {
+  let context = setUp();
+  let iterations = 10 ** 6;
+  let clientGasLimit = 10 ** 8;
+
+  it("when calling a 'view' function should generate an error", async function() {
+    const block = await context.web3.eth.getBlock("latest");
+
+    const isVeryHighGasLimit = clientGasLimit > block.gasLimit && clientGasLimit < Number.MAX_SAFE_INTEGER;
+    assert.strictEqual(isVeryHighGasLimit, true);
+
+    try {
+      // Attempt to run an expensive view function
+      await context.instance.methods.expensiveOperation(iterations).call({ gas: clientGasLimit });
+      assert.fail("Expecting a block gas limit error when executing a expensive 'view' function");
+    } catch (error) {
+      assert.strictEqual(error.message, "Exceeds block gas limit");
```

Comment:
It looks like [Uxio0](https://github.com/trufflesuite/ganache-core/issues/148#issuecomment-435048323) was able to confirm that parity allows for view calls exceeding the the block gas limit.  Were you able to [verify](https://github.com/trufflesuite/ganache-core/issues/148#issuecomment-416342014) if geth does as well?

Reply:
Geth does allows a sender to specify a gasLimit > block.gasLimit. But have not confirmed Parity locally yet.

Reply:
If that's the case ( and assuming parity does as well ) then we definitely need to discuss supporting this feature, as well as the implications, with the team - pinging @davidmurdoch 
@seesemichaelj.  As @benjamincburns alluded, this potentially offers an avenue for a slowloris like 'attack' against an individual node.  Maybe we can explore what 'safe-guard' (if any) geth/parity have in place.

Reply:
It sounds to me like this call _shouldn't_ throw? Assuming we want to align with geth and parity, is this test backwards?

Reply:
The test is backwards in order to expose the different responses from the tool. Will start working on a fix to align Ganache with Geth and Parity.